src/audio/module_adapter/module/generic.c

```diff
@@ -336,6 +336,9 @@ int module_reset(struct processing_module *mod)
 	if (md->state < MODULE_IDLE)
 		return 0;
 #endif
+	/* cancel task if DP task*/
+	if (mod->dev->ipc_config.proc_domain == COMP_PROCESSING_DOMAIN_DP && mod->dev->task)
+		schedule_task_cancel(mod->dev->task);
```

Comment:
do I understand it correctly, that previously DP tasks were terminated from `comp_free()` by calling `schedule_task_free()`? Now `scheduler_dp_task_cancel()` will (potentially) be called twice - from here and from `schedule_task_free()`. Should be safe for now, at least as long as `list_del()` also initialises the list item.

Reply:
Yes., it is called twice, nobody guarantees that cancel had been called before free. 
>> as long as list_del() also initialises the list item

it does: 
```
static inline void list_item_del(struct list_item *item)
{
	item->next->prev = item->prev;
	item->prev->next = item->next;
	list_init(item);
}
```

Reply:
@marcinszkudlinski yes, sorry, that's exactly what I meant: as long as `list_init()` is there, then calling it twice is ok, but if we once decide to remove it since technically it isn't needed there, then this (and probably a couple of other places) will break